owtf/managers/config.py

```diff
@@ -5,73 +5,100 @@
+    :type fallback_file_path: `str`
+    :return: config_map
+    :rtype: dict
+    """
+    file_path = file_path if os.path.isfile(file_path) else fallback_file_path
+    logging.info("Loading Configuration from: {}..".format(file_path))
+    if not os.path.isfile(file_path):
+        # check if the config file exists
+        abort_framework("Config file not found at: {}".format(file_path))
+    try:
+        config_map = yaml.load(FileOperations.open(file_path, 'r'))
+        return config_map
+    except yaml.YAMLError:
+        abort_framework("Error parsing config file at: {}".format(file_path))
 
 
-def load_config_db_file(session, default, fallback):
+def load_general_configs_to_db(session, default, fallback):
     """Load Db config from file
 
-    :param file_path: The path to config file
-    :type file_path: `str`
+    :param session: SQLAlchemy database session
+    :type session: `object`
+    :param default: The fallback path to config file
+    :type default: `str`
+    :param fallback: The path to config file
+    :type fallback: `str`
     :return: None
     :rtype: None
     """
-    file_path = default
-    if not os.path.isfile(file_path):
-        file_path = fallback
-    logging.info("Loading Configuration from: %s.." % file_path)
-    config_parser = parser.RawConfigParser()
-    config_parser.optionxform = str  # Otherwise all the keys are converted to lowercase xD
-    if not os.path.isfile(file_path):  # check if the config file exists
-        abort_framework("Config file not found at: %s" % file_path)
-    config_parser.read(file_path)
-    for section in config_parser.sections():
-        for key, value in config_parser.items(section):
-            old_config_obj = session.query(models.ConfigSetting).get(key)
-            if not old_config_obj or not old_config_obj.dirty:
-                if not key.endswith("_DESCRIP"):  # _DESCRIP are help values
-                    config_obj = models.ConfigSetting(key=key, value=value, section=section)
-                    # If _DESCRIP at the end, then use it as help text
-                    if config_parser.has_option(section, "%s_DESCRIP" % key):
-                        config_obj.descrip = config_parser.get(section, "%s_DESCRIP" % key)
+    config_dump = load_config_file(default, fallback)
+    for section, config_list in get_dict_iter_items(config_dump):
+        for config_map in config_list:
+            try:
+                old_config_obj = session.query(models.ConfigSetting).get(config_map["config"])
+                if not old_config_obj or not old_config_obj.dirty:
+                    config_obj = models.ConfigSetting(
+                        key=config_map["config"],
+                        value=str(config_map["value"]),
+                        section=section)
+                    try:
+                        config_obj.descrip = config_map["description"]
+                    except KeyError:
+                        pass
```

Comment:
`logging.debug` here.

Reply:
I don't think we need a debug log here as we don't have a description of most of the configs and there is no problem with that.

Reply:
Ah. Then let's replace it with `config_obj.descrip = config_map.get("description", "")`

Reply:
Yeah, this would be much better.